src/api/verification_library.py

```diff
@@ -100,3 +111,79 @@ def get_library_item(self, item_name: str) -> Dict:
         }
 
         return item_dict
+
+    def validate_library(self, items: List[str] = []) -> pd.DataFrame:
+        """Check the validity of library items definition. This validity check includes checking the completeness of json specification (against library json schema) and Python verification class definition (against library class interface) and the match between the json and python implementation.
+
+        Args:
+            items: list of str, default []. Library items to validate. By default, summarize all library items loaded at instantiation.
+
+        Returns:
+            pandas.DataFrame that contains validity information of library items.
+        """
+
+        # check `items` type
+        if not isinstance(items, list):
+            logging.error(f"items needs to be list. It cannot be a {type(items)}.")
+            return None
+
+        validity_info = pd.DataFrame(
+            columns=[
+                "library_item_id",
+                "description_brief",
+                "description_detail",
+                "description_index",
+                "description_datapoints",
+                "description_assertions",
+                "description_verification_type",
+                "assertions_type",
+                "datapoints_match",
+            ]
+        )
+        for item in items:
+            validity_info_data = []
+
+            # verify the library.json file
+            for lib_key in library_schema.keys():
+                # check if lib keys exist. "description_detail" key is optional
+                if lib_key not in ["description_detail"] and not self.lib_items[
+                    item
+                ].get(lib_key):
+                    logging.error(
+                        f"{lib_key} doesn't exist. Please make sure to have {lib_key}."
```

Comment:
Detail here, but perhaps we should output the list of necessary keys. That way the user can check its inputs against it.

Reply:
We have the ```validity_info``` return dataframe that includes all the necessary keys as columns. Would this also be a good way to inform users of the necessary keys? Or, should we have a separate return?